Compiling on RHEL8 fails

Hi, it's me again trying to get a CI pipeline for humble@RHEL working.

But there is some issue with Eigen:
```
Starting >>> rsl     
--- stderr: rsl                         
CMake Error at CMakeLists.txt:25 (target_link_libraries):
  Target "rsl" links to:

    Eigen3::Eigen

  but the target was not found.  Possible reasons include:

    * There is a typo in the target name.
    * A find_package call is missing for an IMPORTED target.
    * An ALIAS target is missing.
```

I was able to fix this by just not link the library but add the include folders: https://github.com/christophfroehlich/RSL/commit/8d2bacef568c2ec45705dee04ed8793d3414ca20

I don't have much experience with Eigen, but as it is a header-only library, why should one link against it? 

With the released version of Eigen on rhel8 there are a lot of build warnings -> I also had to change the compiler flags. 

A newer release of Eigen (3.4 for example) hasn't fixed any of those issues.

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,7 @@ project(rsl VERSION 1.1.0 LANGUAGES CXX DESCRIPTION "ROS Support Library")
 
 set(CMAKE_CXX_EXTENSIONS OFF)
 
-find_package(Eigen3 REQUIRED)
+find_package(Eigen3 REQUIRED NO_MODULE)
 find_package(fmt REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(tcb_span REQUIRED)

--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -3,7 +3,7 @@ project(rsl VERSION 1.1.0 LANGUAGES CXX DESCRIPTION "ROS Support Library")
 
 set(CMAKE_CXX_EXTENSIONS OFF)
 
-find_package(Eigen3 REQUIRED NO_MODULE)
+find_package(Eigen3 REQUIRED CONFIG)
 find_package(fmt REQUIRED)
 find_package(rclcpp REQUIRED)
 find_package(tcb_span REQUIRED)

